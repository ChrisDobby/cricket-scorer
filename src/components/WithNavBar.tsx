import * as React from 'react';
import * as styles from './styles';

const containerStyle: React.CSSProperties = {
    ...styles.flexContainerStyle,
    height: '100vh',
};

const componentStyle: React.CSSProperties = {
    ...styles.flexFillStyle,
    marginTop: '8px',
};

const profileItemStyle: React.CSSProperties = {
    marginRight: '8px',
};

const pictureStyle: React.CSSProperties = {
    ...profileItemStyle,
    borderRadius: '35px',
    width: '35px',
    height: '35px',
    boxShadow: '0 4px 6px 0 rgba(0,0,0,.14), 0 4px 5px rgba(0,0,0,-1)',
};

const WithNavBar = (Component: React.ComponentType) => (props: any) => (
    <div style={containerStyle}>
        <nav className="navbar navbar-dark bg-primary">
            <a className="navbar-brand" href="#">Cricket scores live</a>
            {!props.isAuthenticated &&
                <button className="btn btn-outline-light" onClick={props.login}>Register or login</button>}
            {props.isAuthenticated &&
                <span>
                    {props.userProfile &&
                        <React.Fragment>
                            {props.userProfile.picture &&
                                <img src={props.userProfile.picture} style={pictureStyle} />}
                            <span className="navbar-text" style={profileItemStyle}>{props.userProfile.name}</span>
                        </React.Fragment>}
                    <button className="btn btn-outline-light" onClick={props.logout}>Logout</button>
                </span>}
        </nav>

        <div style={componentStyle}>
            <Component {...props} />
        </div>
    </div>);

export default WithNavBar;
