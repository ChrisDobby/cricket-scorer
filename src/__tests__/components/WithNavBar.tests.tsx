import * as React from 'react';
import { render, cleanup } from 'react-testing-library';
import WithNavBar from '../../components/WithNavBar';

describe('WithNavBar', () => {
    beforeEach(cleanup);
    const TestComponent = () => <div />;
    it('should render correctly', () => {
        const TestComponentWithNavBar = WithNavBar({})(TestComponent);
        const { container } = render(<TestComponentWithNavBar />);

        expect(container).toMatchSnapshot();
    });
});
