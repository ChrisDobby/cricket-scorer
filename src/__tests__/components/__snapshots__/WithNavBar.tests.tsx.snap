// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WithNavBar should render correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
      "height": "100vh",
    }
  }
>
  <nav
    className="navbar navbar-dark bg-primary"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      Cricket scores live
    </a>
    <button
      className="btn btn-outline-light"
      onClick={undefined}
    >
      Register or login
    </button>
  </nav>
  <div
    style={
      Object {
        "flex": "1",
        "marginTop": "8px",
        "overflowX": "hidden",
        "overflowY": "auto",
      }
    }
  >
    <div />
  </div>
</div>
`;
